Automattic/woocommerce-payments
# Fix handling of null response in ExPlat client

Fixes #2293

#### Changes proposed in this Pull Request

Fixes handling of null response in ExPlat client

A `null` response from the Explat API indicates the control treatment should be used. This PR makes that more explicit in the client, replacing returned values of `null` with `'control'`.

No tests are added. This class’s tests in Jetpack are not compatible with WCPay.

#### Testing instructions

Add the following code to line 189 in `class-wc-payments.php`.

```
delete_transient( 'abtest_variation_explat_test_jetpack_php_client' );
$abtest = new \WCPay\Experimental_Abtest(
	'<unique-id-here>',
	'woocommerce',
	true
);

var_dump($abtest->get_variation( 'explat_test_jetpack_php_client' ));
die();
```
- Change the value in '<unique-id-here>' to something random
- Open a page in WC Admin and see the `var_dump` output. It should either be `'control'` or `'treatment'`
- Repetitively change the value in '<unique-id-here>' to something else. See that it randomly selects `'control'` or `'treatment'`.





-------------------
